include/pkcs11-helper-1.0/pkcs11h-core.h

```diff
@@ -278,6 +278,45 @@ struct pkcs11h_token_id_s {
 	char label[sizeof (((CK_TOKEN_INFO *)NULL)->label)+1];
 };
 
+
+/**
+ * @brief Provider property enum.
+ */
+enum pkcs11h_provider_property_type_e {
+	PKCS11H_PROVIDER_PROPERTY_LOCATION = 0, /*!< Provider location. NULL by default */
+	PKCS11H_PROVIDER_PROPERTY_ALLOW_PROTECTED_AUTH = 1, /*!< Allow this provider to use protected authentication. False by default */
+	PKCS11H_PROVIDER_PROPERTY_MASK_PRIVATE_MODE = 2, /*!< Provider private mode @ref PKCS11H_PRIVATEMODE_MASK override. @ref PKCS11H_PRIVATEMODE_MASK_AUTO by default */
+	PKCS11H_PROVIDER_PROPERTY_SLOT_EVENT_METHOD = 3, /*!< Provider slot event @ref PKCS11H_SLOTEVENT_METHOD method. @ref PKCS11H_SLOTEVENT_METHOD_AUTO by default */
+	PKCS11H_PROVIDER_PROPERTY_SLOT_POLL_INTERVAL = 4, /*!< Slot event poll interval (If in polling mode), specify 0 for default. */
+	PKCS11H_PROVIDER_PROPERTY_CERT_IS_PRIVATE = 5, /*!< Provider's certificate access should be done after login. False by default */
+	PKCS11H_PROVIDER_PROPERTY_INIT_ARGS = 6, /*!< Provider initialize arguments. NULL by default */
+};
+
+/**
+ * @brief Provider property identidier.
+ * Identivier sets up pkcs11h_provider_property_t type indide pkcs11h_setProviderProperty.
+ */
+typedef enum pkcs11h_provider_property_type_e pkcs11h_provider_property_type_t;
+
+/**
+ * @brief Provider property type union.
+ */
+union pkcs11h_provider_property_u {
+	const char * provider_location;
+	PKCS11H_BOOL allow_protected_auth;
+	unsigned mask_private_mode;
+	unsigned slot_event_method;
+	unsigned slot_poll_interval;
+	PKCS11H_BOOL cert_is_private;
+	CK_C_INITIALIZE_ARGS_PTR init_args;
```

Comment:
can you please split this patch and the addition of init_args functionality? I can see below that you have dropped existing functionality. let's merge only the set property first.